sdk/common/error_writer.go

```diff
@@ -0,0 +1,25 @@
+package common
+
+import "os"
+
+type ErrorWriter struct {
```

Comment:
I don't think you want to define `ErrorWriter` as a separate `struct` since it still needs to expose all the methods of `os.File`.
I would define it as

```go
type ErrorWriter os.File
```

That way every `ErrorWriter` is-a `os.File` but you can define additional methods on `ErrorWriter`.
It also makes the creation of `ErrorWriter` simpler because it becomes just
```
common.ErrorWriter(os.Stderr)
```